src/main/java/com/pirimid/cryptotrade/helper/exchange/coinbase/EXC_Coinbase.java

```diff
@@ -0,0 +1,187 @@
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Instant;
+import java.util.List;
+
+@Component
+public class EXC_Coinbase implements EXC_Parent {
+    @Value("${api.exchange.coinbase.baseurl}")
+    private String baseUrl;
+
+    private ResponseEntity<String> apiCallerOpen(String uri) throws IOException, InterruptedException {
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(uri))
+                .setHeader("Content-Type", "application/json")
+                .method("GET", HttpRequest.BodyPublishers.noBody())
+                .build();
+        HttpResponse<String> response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return new ResponseEntity<String>(response.body(), headers, response.statusCode());
+    }
+
+    private ResponseEntity<String> apiCallerRestricted(String uri, String method, String apiKey, String passphrase, String signature, String timestamp, String body) throws IOException, InterruptedException {
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(uri))
+                .setHeader("Content-Type", "application/json")
+                .setHeader("CB-ACCESS-KEY", apiKey)
+                .setHeader("CB-ACCESS-PASSPHRASE", passphrase)
+                .setHeader("CB-ACCESS-SIGN", signature)
+                .setHeader("CB-ACCESS-TIMESTAMP", timestamp)
+                .method(method, HttpRequest.BodyPublishers.ofString(body))
+                .build();
+        HttpResponse<String> response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return new ResponseEntity<String>(response.body(), headers, response.statusCode());
+    }
+
+    @Override
+    public List<SymbolResDTO> getPairs() {
+        SymbolResCoinbaseDTO symbolResCoinbaseDTO;
+        Gson gson = new Gson();
+        List<SymbolResCoinbaseDTO> symbolResCoinbaseDTOS;
+        List<SymbolResDTO> symbolResDTOS;
+        try {
+            ResponseEntity<String> response =  apiCallerOpen(baseUrl + "/products");
+            Type SymbolListType = new TypeToken<List<SymbolResCoinbaseDTO>>(){}.getType();
+            symbolResCoinbaseDTOS = gson.fromJson(response.getBody(),SymbolListType);
+            symbolResDTOS = CoinbaseUtil.getPairsResDTO(symbolResCoinbaseDTOS);
+            return symbolResDTOS;
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        return null;
```

Comment:
If null means error then can we return badRequest in response?

Reply:
yes, will do that